### ISSUE TITLE
Android CI hangs

### ISSUE THREAD

**Issue by niyaznigmatullin** [uutils/coreutils#3878]
After updating `chrono` from `0.4.19` to `0.4.22` android CI hangs from time to time. If getting back to `0.4.19` it stops doing that, and works fine. I tested that on my fork, and on android emulator on local computer.

The changes happened in `chrono` between these versions are:
- new timezone detection introduced
- fixed timezone detection for different platforms and specifically for Android
- new dependency `iana_time_zone` was used for that

I spent some time to investigate why it fails in a local emulator, almost every time it's the `pr` tests that don't terminate, and the `pr` binary is waiting in `do_exit()`, and the test is waiting in `libc::syscall -> __kernel_vsyscall`. So I was testing only `test_pr::` tests. And hanging happens so indefinitely, for instance, if you call binary normally trying to get its exit code, it hangs often, if you call it in background with `&` not trying to get its exit code, it doesn't hang.

I would say that probably some undefined behavior happens somewhere in some library, or in our code.

**Comment by sylvestre** [comment]
well spotted

Do you have the full backtrace?

**Comment by niyaznigmatullin** [comment]
I explored something else. I got another result this time.

So we have two processes alive while it hangs: the root testing binary (25942), the process that was forked from root for a single test (25946). And the `pr` process (25952) was already terminated, but others were unaware of that.

This is their strace output: 
[strace.tar.gz](https://github.com/uutils/coreutils/files/9435754/strace.tar.gz)

And in 25952 `exec` wasn't even called. The last syscall of 25952 is `futex`. 25946, its parent, were trying to read from pipe the output of 25952, but was failing in the end. 25942 was just waiting when 25946 terminates.
